src/backend/joanie/core/models/products.py

```diff
@@ -0,0 +1,253 @@
+            .exclude(state__in=[enums.ORDER_STATE_CANCELED])
+            .exists()
+        ):
+            raise exceptions.OrderAlreadyExists("Order already exist")
+
+        # first create an order
+        order = Order.objects.create(product=self, owner=user)
+        for resource_link in resource_links:
+            # Course run have to be available for the product
+            course_run = self.course_runs.get(resource_link=resource_link)
+            # associate each course run selected to the order
+            order.course_runs.add(course_run)
+            # then create enrollment for each course run
+            enrollment = Enrollment.objects.create(course_run=course_run, order=order)
+            # now we can enroll user to LMS course run
+            lms = LMSHandler.select_lms(resource_link)
+            # if no lms found we set enrollment and order to failure state
+            # this issue could be due to a bad setting or a bad resource_link filled,
+            # so we need to log this error to fix it quickly to joanie side
+            if lms is None:
+                order.state = enums.ORDER_STATE_FAILED
+                order.save()
+                enrollment.state = enums.ENROLLMENT_STATE_FAILED
+                enrollment.save()
+                logger.error(
+                    "No LMS configuration found for resource link: %s", resource_link
+                )
+            # now enroll user to lms course run and change joanie enrollment state to 'in_progress'
+            else:
+                lms_enrollment = lms.set_enrollment(user.username, resource_link)
+                if lms_enrollment["is_active"]:
+                    enrollment.state = enums.ENROLLMENT_STATE_IN_PROGRESS
+                    enrollment.save()
+        return order
+
+
+class ProductCourseRunPosition(models.Model):
+    """
+    ProductCourseRunPosition model allows to define order of each course runs to follow
+    for a product.
+    Some course runs could have the same position for a product (various sessions for example).
+    Validation of only one of them will be enough to pass to the next course run or certification.
+    """
+
+    product = models.ForeignKey(
+        Product,
+        verbose_name=_("product"),
+        related_name="course_runs_positions",
+        on_delete=models.CASCADE,
+    )
+    position = models.PositiveSmallIntegerField(_("position in product"))
+    course_run = models.ForeignKey(
+        courses_models.CourseRun,
+        verbose_name=_("course run"),
+        related_name="positions",
+        on_delete=models.RESTRICT,
+    )
+
+    class Meta:
+        db_table = "joanie_product_course_run_position"
+        verbose_name = _("Position of course runs in products")
+        verbose_name_plural = _("Positions of course runs in products")
+        unique_together = ("product", "course_run", "position")
+
+    def __str__(self):
+        return f"{self.product}: {self.position}/ {self.course_run}]"
```

Comment:
should you initialize the position field upon object creation based on existing `ProductCourseRunPosition` objects?

Reply:
yes good idea!